src/v2/Apps/Fair/Components/FairCollections/FairCollection.tsx

```diff
@@ -18,11 +25,32 @@ const CARD_IMAGE_SIZES = [
 
 interface FairCollectionProps {
   collection: FairCollection_collection
+  fairID: string // needed for analytics
+  fairSlug: string // needed for analytics
+  carouselIndex: number // needed for analytics
```

Comment:
The needed for analytics comments freak me out a bit. It'd be nice if there was a way to provide context to the tracking hook without having to prop drill like this but I see that it's some library.

ID/slug can be part of the fragment for this component rather than requiring them from the parent, though.

Reply:
Agreed.

Reply:
@sweir27 / @dzucconi - got distracted before I could respond to this yesterday (before merge) -- we do [have a new `AnalyticsContext`](https://github.com/artsy/force/blob/ebe49899c529c4c4d27206a99dd76a77a55c3c0e/src/v2/Artsy/Analytics/AnalyticsContext.tsx)! @eessex added it just before vacation. I think we should def leverage it for things like this, though the API might need to slightly change to account for more ephemeral values.

Reply:
Whoa, cool!

Reply:
yes we should definitely use this - @sweir27 happy to pair on this, there are some examples implementing it in here: https://github.com/artsy/force/pull/6213

Reply:
@eessex - looking through that PR reminds me, we should pick up the discussion about moving the provider to `Boot` and providing a mechanism for _updating_ it versus reinitializing by adding it to to multiple branches of the react tree. For the most part, the common ancestor for all of these components is Boot, so it can likely be added one time and then augmented through time with new values.